acceptance/install/install_test.go

```diff
@@ -0,0 +1,221 @@
+	assert.Equal(t, "", stderr)
+	assert.Equal(t, 0, exitCode)
+
+	// Tear Down
+	removeTemplate(templatePath, templatePkgName)
+}
+
+type templateData struct {
+	name          string
+	listExpRegex  string
+	expectedFiles []string
+}
+
+func Test_PctInstall_InstallsTo_DefinedTemplatePath(t *testing.T) {
+	testutils.SkipAcceptanceTest(t)
+
+	// Setup
+	templatePath := testutils.GetTmpDir(t)
+
+	templatePkgs := []templateData{
+		{
+			name:         "additional-project",
+			listExpRegex: "Additional\\sProject\\s+\\|\\sadditional-project\\s+\\|\\sproject",
+			expectedFiles: []string{
+				"pct-config.yml",
+				"content/empty.txt",
+				"content/goodfile.txt.tmpl",
+			},
+		},
+		{
+			name:         "good-project",
+			listExpRegex: "Good\\sProject\\s+\\|\\sgood-project\\s+\\|\\sproject",
+			expectedFiles: []string{
+				"pct-config.yml",
+				"content/empty.txt",
+				"content/goodfile.txt.tmpl",
+			},
+		},
+	}
+
+	for _, template := range templatePkgs {
+		// Setup
+		templatePkgPath, _ := filepath.Abs(fmt.Sprintf("../../acceptance/install/testdata/%v.tar.gz", template.name))
+		installCmd := fmt.Sprintf("install %v --templatepath %v", templatePkgPath, templatePath)
+
+		// Exec
+		stdout, stderr, exitCode := testutils.RunPctCommand(installCmd, "")
+
+		// Assert
+		assert.Contains(t, stdout, fmt.Sprintf("Template installed to %v", filepath.Join(templatePath, template.name)))
+		assert.Equal(t, "", stderr)
+		assert.Equal(t, 0, exitCode)
+	}
+
+	for _, template := range templatePkgs {
+		// Setup
+		listCmd := fmt.Sprintf("new --list --templatepath %v", templatePath)
+
+		// Assert
+		for _, file := range template.expectedFiles {
+			assert.FileExists(t, filepath.Join(templatePath, template.name, file))
+		}
+
+		stdout, stderr, exitCode := testutils.RunPctCommand(listCmd, "")
+
+		assert.Regexp(t, template.listExpRegex, stdout)
+		assert.Equal(t, "", stderr)
+		assert.Equal(t, 0, exitCode)
+	}
+
+	// Tear Down
+	for _, template := range templatePkgs {
+		removeTemplate(templatePath, template.name)
+	}
+}
+
+func Test_PctInstall_Errors_When_NoTemplatePkgDefined(t *testing.T) {
+	testutils.SkipAcceptanceTest(t)
+
+	// Setup
+	installCmd := "install"
+
+	// Exec
+	stdout, stderr, exitCode := testutils.RunPctCommand(installCmd, "")
+
+	// Assert
+	assert.Contains(t, stdout, "Must specify a path to a template package")
+	assert.Equal(t, "exit status 1", stderr)
+	assert.Equal(t, 1, exitCode)
+}
+
+func Test_PctInstall_Errors_When_TemplatePkgNotExist(t *testing.T) {
+	testutils.SkipAcceptanceTest(t)
+
+	// Setup
+	templatePath := "/path/to/nowhere"
+	templatePkgName := "good-project"
+	templatePkgPath, _ := filepath.Abs(fmt.Sprintf("%v/%v.tar.gz", templatePath, templatePkgName))
+	installCmd := fmt.Sprintf("install %v", templatePkgPath)
+
+	// Exec
+	stdout, stderr, exitCode := testutils.RunPctCommand(installCmd, "")
+
+	// Assert
+	assert.Contains(t, stdout, fmt.Sprintf("No template package at %v", templatePkgPath))
```

Comment:
too many variables, making this test hard to read. These could be strings. Copy paste if needed.

Reply:
Please review the rest of these functions for the same sort of issue.